src/System.Management.Automation/engine/parser/tokenizer.cs

```diff
@@ -3258,10 +3257,36 @@ private static bool TryGetNumberValue(string strNum, bool hex, bool real, char s
                         return false;
                     }
 
+                    if (suffix == 'u' || suffix == 'U')
+                    {
+                        if (UInt32.TryParse(strNum, style, NumberFormatInfo.InvariantInfo, out uint u))
+                        {
+                            ulong testresult = u * (ulong)multiplier;
+                            if (testresult > UInt32.MaxValue)
+                            {
+                                result = testresult;
+                            }
+                            else
+                            {
+                                result = (uint)testresult;
+                            }
+                            return true;
+                        }
+                        else if (UInt64.TryParse(strNum, style, NumberFormatInfo.InvariantInfo, out ulong u2))
+                        {
+                            result = u2 * (ulong)multiplier;
```

Comment:
We could remove UInt32.TryParse and use only UInt64.TryParse and cast based on MaxValue.

Reply:
Hmm. That is true. And considering that I'm already half doing that having to account for the multiplier... yeah, that makes sense. I'll do that, thanks!